app/models/miq_event_handler/runner.rb

```diff
@@ -1,2 +1,39 @@
 class MiqEventHandler::Runner < MiqQueueWorkerBase::Runner
+  def artemis?
+    Settings.prototype.queue_type == 'artemis'
+  end
+
+  def do_before_work_loop
+    if artemis?
+      topic_options = {
+        :service     => "events",
+        :persist_ref => "event_handler"
+      }
+
+      # this block is stored in a lambda callback and is executed in another thread once a msg is received
+      # FIXME: is the subscription stored in artemis?
+      # Would it hurt to subscribe again on a second startup of the worker?
```

Comment:
@bzwei @kbrock any idea on this? and also the question in `before_exit` below? I'm not sure where the subscription is stored. Should the `EventHandler` worker subscribe on startup and unsubscribe on exit?

Reply:
@durandom Yes the subscription is stored in artemis. If you close the client before exit, it is automatically unsubscribed. If you don't close it will eventually timeout at the server side, but it is not a good practice.

Reply:
@bzwei so in this specific case, I'd close the connection (although it could be shared) - but as we are running as a worker, we are the only process that uses the connection

Reply:
There is `unsubscribe` method that we already delegate to stomp client. We need to expose it in the interface level. A future PR.
Anyway closing connection is still needed on exit.